src/components_redux/Input.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from "react";
+
+class Input extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.openNav = this.openNav.bind(this);
+    //this.closeNav = this.closeNav.bind(this);
+  }
+
+  componentDidMount() {
+    document.addEventListener("click", this.closeNav);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener("click", this.closeNav);
+  }
+
+  openNav() {
+    const style = { width: 550 };
+    this.setState({ style });
+    document.body.style.backgroundColor = "rgba(0,0,0,0.4)";
+    document.addEventListener("click", this.closeNav);
```

Comment:
closeNav is not defined.

Inside of react components, state and click handlers from react should be used. Do not use your own event listeners if not really needed.

Reply:
I'll remove this.